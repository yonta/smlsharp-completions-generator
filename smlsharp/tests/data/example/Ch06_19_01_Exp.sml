fun fact 0 = 1
  | fact n = n * fact (n - 1);
fun factorial n = fact n;
fun fib (n:int) = n;

fact 4 :: 4 + 4 :: (if factorial 1 = 0 then nil else [1,2,3]); 
[1.1, Math.pi, Math.sqrt 2.0]; 
"I"::"became"::"fully"::"operational"::"on"::"April 2, 2012"::nil; 
[factorial, fib]; 
[#"S", #"M", #"L", #"#"]; 
implode it; 
explode it; 
