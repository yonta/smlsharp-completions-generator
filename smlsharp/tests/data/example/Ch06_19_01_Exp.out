val fact = fn : int -> int

val factorial = fn : int -> int

val fib = fn : int -> int

val it = [24, 8, 1, 2, 3] : int list

val it = [1.1, 3.14159265359, 1.41421356237] : real list

val it =
  ["I", "became", "fully", "operational", "on", "April 2, 2012"] : string list

val it = [fn, fn] : (int -> int) list

val it = [#"S", #"M", #"L", #"#"] : char list

val it = "SML#" : string

val it = [#"S", #"M", #"L", #"#"] : char list
